GitHub integration docs (#3638)
astronomer/docs: astro/deploy-github-integration.md

---
title: 'Deploy code with the Astro GitHub integration'
id: deploy-github-integration

> description: Learn about the different ways you can deploy code to Astro.

Reviewer: ```suggestion
description: Learn how to automatically deploy Apache Airflow code to Astro from GitHub with a native integration.
```

unlisted: true
---

> 

Reviewer: @lzdanski Can you add the standard Private Preview banner on top here?

> Integrating GitHub with Astro is the fastest way to implement CI/CD for Apache Airflow and deploy code to Astro. Astro’s automatic deploy system eliminates the need to implement GitHub Actions and gives you greater visibility into the code you’re running on Astro.

Reviewer: ```suggestion
Astronomer's native GitHub integration is the fastest way to implement CI/CD for Apache Airflow and deploy code to Astro. Astro’s automatic deploy system eliminates the need to implement GitHub Actions and gives you greater visibility into the code you’re running on Astro.
```

> To deploy code through an integrated GitHub repository, you first connect a GitHub repository with your Airflow code to an Astro Workspace. Then, you map a Git branch in that repository to an Astro Deployment. When a pull request is merged into your mapped branch, your code is automatically deployed to Astro.

Reviewer: ```suggestion
To deploy code through an integrated GitHub repository, you first connect a GitHub repository with your Astro project to an Astro Workspace. Then, you map a Git branch in that repository to an Astro Deployment. When a pull request is merged into your mapped branch, your code is automatically deployed to Astro.
```

[DIAGRAM]

> 

Reviewer: ```suggestion
```

> Compared to deploying code manually, using the Astro CLI, or through a custom CI/CD process, using the Astro’s GitHub integration:

Reviewer: ```suggestion
Compared to deploying code manually using the Astro CLI or through a custom CI/CD process, using Astro’s GitHub integration:
```

- Write descriptive commit messages or pull request descriptions for any project changes. These messages appear in the Astro UI under **Deploy History** to give your team more context about each deploy.

## Limitations

The Astro GitHub integration is not supported if any of the following are true:

- You build a custom Astro Runtime image as part of your deploy process. However, using the `-base` distribution of Astro Runtime is supported.
- You need to deploy images from an external or private Docker registry.
- You use the `--image-only` flag in the Astro CLI, or you need to exclude DAGs from specific deploys. This is usually the case if you have a multiple-repository deploy strategy as described in [Develop a CI/CD strategy](https://docs.astronomer.io/astro/set-up-ci-cd#multiple-repositories).
- Your repository is hosted in an on-premises GitHub Enterprise server. To deploy code from an on-premises repository, see [Private network CI/CD templates](https://docs.astronomer.io/astro/ci-cd-templates/github-actions-private-network).

> - There are currently no checks to prevent you from downgrading your version of Astro Runtime. Astronomer recommends only downgrading Deployments using rollbacks.

Reviewer: ```suggestion
- There are currently no checks to prevent you from downgrading your version of Astro Runtime. Astronomer recommends only downgrading Deployments using [rollbacks](/deploy-history.md/roll-back-to-a-past-deploy)
```
@lzdanski Check me on this link

There is currently no limit or additional charge for deploying code to Astro with the GitHub integration, but Astronomer might reach out to discuss your usage if you trigger an unusually large number of deploys.

## Prerequisites

- Workspace Admin permissions on an Astro Workspace.

> - A GitHub repository that contains an Astro project. The Astro project can exist at any level in your repository.

Author: ```suggestion
- A GitHub repository that contains an [Astro project](https://docs.astronomer.io/astro/cli/develop-project#create-an-astro-project). The Astro project can exist at any level in your repository.
```

> - A GitHub account with read and write permissions to the repositories containing your Astro project. If you don't have these permissions, send a request GitHub repository administrator when prompted by Astro.

Reviewer: ```suggestion
- A GitHub account with read and write permissions to the repositories containing your Astro project. If you don't have these permissions, send a request to your GitHub repository administrator when prompted by Astro.
```

- At least one Astro Deployment.
- At least one GitHub branch that you want to deploy to Astro.

## Connect a GitHub repository to a Workspace

> Before you begin, ensure that you’re logged in on GitHub with permissions to read code from the repository where you want to deploy code.

Reviewer: ```suggestion
Before you begin, ensure that you’re logged in to GitHub with permissions to read code from the repository where you want to deploy code.
```

1. In your Workspace, click **Workspace Settings** > **Git Deploys.**
2. Click **Install GitHub Application.** A window appears instructing you to authorize the **Astro App** on your personal GitHub account. Follow the prompts to authorize the application.
3. Return to the Astro UI. From the **Git Deploys** screen, click **Connect Repository**. 

> 4. Select the Organization that contains the repository you want to integrate with Astro, then click **Continue**. A new window prompts you to allow **Astro App** to access either all repositories or specific repositories within your GitHub Organization. Choose which repositories you want to enable the app for, then click **Request.**

Reviewer: ```suggestion
4. Select the Organization that contains the repository you want to integrate with Astro, then click **Continue**. A new window prompts you to allow **Astro App** to access either all repositories or specific repositories within your GitHub Organization. Astronomer recommends **Only selected repositories**.
5. Choose which repositories you want to enable the app for, then click **Request.**
```

    
    ![The GitHub authorization screen for connecting a repository to GitHub. GitHub requests for the Astro App to connect to at least one of your repositories](/img/docs/request-astro-app.png)
    
5. Configure the following fields:
   
    - **Repository:** Select the repository you want to integrate with Astro.
    - **Astro Project Path:** Specify the path to your Astro project, up to and including the Astro project folder. For example, `/myorg/myprojects/my-astro-project`
  
6. Click **Connect Repository**.

> 7. Map specific branches in your repository to Deployments in your Workspace. When you map a branch to a Deployment, any commits to that branch trigger a code deploy to Astro. For example, you can map a development branch and your production branch to separate Deployments so that bugs in development don’t affect your production data pipelines.

Reviewer: ```suggestion
6. Configure the following fields:
   
    - **Repository:** Select the repository you want to integrate with Astro.
    - **Astro Project Path:** Specify the path to your Astro project, up to and including the Astro project folder. For example, `/myorg/myprojects/my-astro-project`
  
7. Click **Connect Repository**.
8. Map specific branches in your repository to Deployments in your Workspace. When you map a branch to a Deployment, any commits to that branch trigger a code deploy to Astro. For example, you can map a development branch and your production branch to separate Deployments so that bugs in development don’t affect your production data pipelines.
```

> Any commits to your mapped branches will now trigger a code deploy to the related Deployment.

Reviewer: ```suggestion
Any commits to your mapped branches will now trigger a code deploy to the corresponding Deployment.
```

## Deploy from GitHub to Astro

> To deploy code from your GitHub repository to Astro, make a commit to one of your mapped branches. This commit triggers the Astro App to deploy your Astro project.

Reviewer: ```suggestion
To deploy code from your GitHub repository to Astro, you can either:

- Make a direct commit to one of your mapped branches.
- Merge a pull request against one of our mapped branches.

Both of these actions triggers the Astro App to deploy your Astro project to the mapped Astro Deployment. When DAG-only deploys are enabled, your GitHub repository triggers:

- A DAG-only deploy if only your DAGs are changed. 
- A full project image deploy if you change a configuration in your project.

If DAG-only deploys are disabled, all code changes will trigger a full project image deploy. To learn more about DAG-only deploys, see [Deploy DAGs to Astro](deploy-dags.md).
```

We didn't take into account that this feature could be disabled

> - If you only change your DAGs, your GitHub repository will trigger a DAG-only deploy.
> - If you change a configuration in your project, your GitHub repository will trigger a full project image deploy.

Reviewer 2: ```suggestion
- If you only change your DAGs, your GitHub repository triggers a DAG-only deploy.
- If you change a configuration in your project, your GitHub repository triggers a full project image deploy.
```

> - If you only change your DAGs, your GitHub repository triggers a DAG-only deploy.
> - If you change a configuration in your project, your GitHub repository triggers a full project image deploy.

Reviewer: ```suggestion
```

You can check the status of your deploy in the Astro UI. 

## Review code deploys from the Astro UI

When you trigger a code deploy by committing a change to one of your mapped branches, details about the code deploy appear in the Astro UI. For past and currently running deploys, you can review:

> - Who triggered the deploy
> - When the deploy was triggered
> - Which pull request or commit triggered the deploy

Reviewer: ```suggestion
- Which user triggered the deploy.
- At what time the deploy was triggered.
- Which pull request or commit triggered the deploy. This includes the Git commit description and a link to your Pull Request if applicable.
```

---
title: 'Deploy code with the Astro GitHub integration'
id: deploy-github-integration
description: Learn how to automatically deploy Apache Airflow code to Astro from GitHub with a built-in integration.
unlisted: true
---

:::privatepreview
:::

> Astronomer's built-in GitHub integration is the fastest way to implement CI/CD for Apache Airflow and deploy code to Astro. Astro’s automatic deploy system eliminates the need to implement GitHub Actions and gives you greater visibility into the code you’re running on Astro.

Reviewer 2: ```suggestion
Astronomer's built-in GitHub integration is the fastest way to implement CI/CD for Apache Airflow and deploy code to Astro. Astro’s automatic deploy system eliminates both the need to implement GitHub Actions and gives you greater visibility into the code you’re running on Astro.
```

## Prerequisites

- Workspace Admin permissions on an Astro Workspace.
- A GitHub repository that contains an [Astro project](https://docs.astronomer.io/astro/cli/develop-project#create-an-astro-project). The Astro project can exist at any level in your repository.
- A GitHub account with read and write permissions to the repositories containing your Astro project. If you don't have these permissions, send a request to your GitHub repository administrator when prompted by Astro.
- At least one Astro Deployment.
- At least one GitHub branch that you want to deploy to Astro.

## Connect a GitHub repository to a Workspace

Before you begin, ensure that you’re logged in to GitHub with permissions to read code from the repository where you want to deploy code.

1. In your Workspace, click **Workspace Settings** > **Git Deploys.**
2. Click **Install GitHub Application.** A window appears instructing you to authorize the **Astro App** on your personal GitHub account. Follow the prompts to authorize the application.
3. Return to the Astro UI. From the **Git Deploys** screen, click **Connect Repository**.

> 4. Select the Organization that contains the repository you want to integrate with Astro, then click **Continue**. A new window prompts you to allow **Astro App** to access either all repositories or specific repositories within your GitHub Organization. Astronomer recommends **Only selected repositories**.

Reviewer 2: ```suggestion
4. Select the Organization that contains the repository you want to integrate with Astro, then click **Continue**. A new window prompts you to allow **Astro App** to access either all repositories or specific repositories within your GitHub Organization. Astronomer recommends **Only select repositories**.
```

5. Choose which repositories you want to enable the app for, then click **Request.**

    ![The GitHub authorization screen for connecting a repository to GitHub. GitHub requests for the Astro App to connect to at least one of your repositories](/img/docs/request-astro-app.png)

> 5. Configure the following fields:

Reviewer 2: ```suggestion
6. Configure the following fields:
```

    - **Repository:** Select the repository you want to integrate with Astro.
    - **Astro Project Path:** Specify the path to your Astro project, up to and including the Astro project folder. For example, `/myorg/myprojects/my-astro-project`

> 6. Click **Connect Repository**.
> 7. Map specific branches in your repository to Deployments in your Workspace. When you map a branch to a Deployment, any commits to that branch trigger a code deploy to Astro. For example, you can map a development branch and your production branch to separate Deployments so that bugs in development don’t affect your production data pipelines.

Reviewer 2: ```suggestion
7. Click **Connect Repository**.
8. Map specific branches in your repository to Deployments in your Workspace. When you map a branch to a Deployment, any commits to that branch trigger a code deploy to Astro. For example, you can map a development branch and your production branch to separate Deployments, so that bugs in development don’t affect your production data pipelines.
```

Any commits to your mapped branches will now trigger a code deploy to the corresponding Deployment.

## Deploy from GitHub to Astro

To deploy code from your GitHub repository to Astro, you can either:

- Make a direct commit to one of your mapped branches.
- Merge a pull request against one of our mapped branches.

Both of these actions triggers the Astro App to deploy your Astro project to the mapped Astro Deployment. When DAG-only deploys are enabled, your GitHub repository triggers:

- A DAG-only deploy if only your DAGs are changed. 
- A full project image deploy if you change a configuration in your project.

If DAG-only deploys are disabled, all code changes will trigger a full project image deploy. To learn more about DAG-only deploys, see [Deploy DAGs to Astro](deploy-dags.md).

You can check the status of your deploy in the Astro UI.

## Review code deploys from the Astro UI

When you trigger a code deploy by committing a change to one of your mapped branches, details about the code deploy appear in the Astro UI. For past and currently running deploys, you can review:

- Which user triggered the deploy.
- At what time the deploy was triggered.

> - Which pull request or commit triggered the deploy. This includes the Git commit description and a link to your Pull Request if applicable.

Reviewer 2: ```suggestion
- Which pull request or commit triggered the deploy. This includes the Git commit description and a link to your pull request if applicable.
```
